File: tone_measure/get_data.r
category <- c("positive", "negative", "uncertainty",
                "litigious", "modal_strong", "modal_weak")

base_url <- "http://www3.nd.edu/~mcdonald/Data/Finance_Word_Lists"

url <- file.path(base_url,
                 c("LoughranMcDonald_Positive.csv",
                   "LoughranMcDonald_Negative.csv",
                   "LoughranMcDonald_Uncertainty.csv",
                   "LoughranMcDonald_Litigious.csv",
                   "LoughranMcDonald_ModalStrong.csv",
                   "LoughranMcDonald_ModalWeak.csv"))

df <- data.frame(category, url, stringsAsFactors=FALSE)

getWords <- function(url) {
    words <- read.csv(url, as.is=TRUE)
    paste(words[,1], collapse=",")
}
df$words <- unlist(lapply(df$url, getWords))

library(readr)
write_csv(df, path = "lm_words.csv")

File: tone_measure/get_data.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Tue May  7 20:13:01 2019

@author: carrascod
"""

import pandas as pd
import requests
import io

base_url = "http://www3.nd.edu/~mcdonald/Data/Finance_Word_Lists"

url  = ["LoughranMcDonald_Positive.csv",
                   "LoughranMcDonald_Negative.csv",
                   "LoughranMcDonald_Uncertainty.csv",
                   "LoughranMcDonald_Litigious.csv",
                   "LoughranMcDonald_ModalStrong.csv",
                   "LoughranMcDonald_ModalWeak.csv"]

category = ["positive", "negative", "uncertainty",
            "litigious", "modal_strong", "modal_weak"]

r = requests.post(base_url+'/'+url[0])

print(r)


if r.ok:
    data = r.content.decode('utf8')
    df = pd.read_csv(io.StringIO(data))

df.to_csv("lm_words_pythonb.csv")